qiskit/transpiler/passes/crosstalk_adaptive_schedule.py

```diff
@@ -0,0 +1,667 @@
+            g_first = self.first_gate_on_qubit[q]
+            finish_time = self.gate_start_time[g_last] + self.gate_duration[g_last]
+            start_time = self.gate_start_time[g_first]
+            if q in self.measured_qubits:
+                self.opt.add(self.measure_start >= finish_time)
+                self.opt.add(self.qubit_lifetime[q] == self.measure_start - start_time)
+            else:
+                # All qubits get measured simultaneously whether or not they need a measurement
+                self.opt.add(self.measure_start >= finish_time)
+                self.opt.add(self.qubit_lifetime[q] == finish_time - start_time)
+
+    def objective_function(self):
+        """
+        Objective function is a weighted combination of gate errors and decoherence errors
+        """
+        self.fidelity_terms = [self.gate_fidelity[gate] for gate in self.gate_fidelity]
+        self.coherence_terms = []
+        for q in self.qubit_lifetime:
+            val = -self.qubit_lifetime[q]/min(self.bp_t1_time[q], self.bp_t2_time[q])
+            self.coherence_terms.append(val)
+
+        all_terms = []
+        for item in self.fidelity_terms:
+            all_terms.append(self.weight_factor*item)
+        for item in self.coherence_terms:
+            all_terms.append((1-self.weight_factor)*item)
+        self.opt.maximize(Sum(all_terms))
+
+    def r2f(self, val):
+        """
+        Convert Z3 Real to Python float
+        """
+        return float(val.as_decimal(16).rstrip('?'))
+
+    def extract_solution(self):
+        """
+        Extract gate start and finish times from Z3 solution
+        """
+        self.m = self.opt.model()
+        result = {}
+        for tmpg in self.gate_start_time:
+            start = self.r2f(self.m[self.gate_start_time[tmpg]])
+            dur = self.r2f(self.m[self.gate_duration[tmpg]])
+            result[tmpg] = (start, start+dur)
+        return result
+
+    def solve_optimization(self):
+        """
+        Setup and solve a Z3 optimization for finding the best schedule
+        """
+        self.create_z3_vars()
+        self.basic_bounds()
+        self.scheduling_constraints()
+        self.fidelity_constraints()
+        self.coherence_constraints()
+        self.objective_function()
+        # Solve step
+        self.opt.check()
+        # Extract the schedule computed by Z3
+        result = self.extract_solution()
+        return result
+
+    def check_dag_dependancy(self, gate1, gate2):
+        """
+        gate2 is a DAG dependent of gate1 if it is a descendant of gate1
+        """
+        return gate2 in self.dag.descendants(gate1)
+
+    def check_xtalk_dependency(self, t_1, t_2):
+        """
+        Check if two gates have a crosstalk dependency
+        """
+        g_1 = t_1[0]
+        s_1 = t_1[1]
+        f_1 = t_1[2]
+        g_2 = t_2[0]
+        s_2 = t_2[1]
+        f_2 = t_2[2]
+        # We don't consider single qubit xtalk
```

Comment:
I think this is an important note and should probably go in the pass docstring itself.

Reply:
I've added it to the init method doc string.